src/main/java/org/cyclonedx/util/ExtensionSerializer.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is part of CycloneDX Core (Java).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.cyclonedx.util;
+
+import java.io.IOException;
+
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.dataformat.xml.ser.ToXmlGenerator;
+import org.cyclonedx.model.ExtensibleType;
+import org.cyclonedx.model.Extension;
+import org.cyclonedx.model.Extension.ExtensionType;
+import org.cyclonedx.model.vulnerability.Rating;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Advisory;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Cwe;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Recommendation;
+
+public class ExtensionSerializer
+    extends StdSerializer<Extension>
+{
+  public ExtensionSerializer() {
+    this(null);
+  }
+
+  public ExtensionSerializer(final Class t) {
+    super(t);
+  }
+
+  @Override
+  public void serialize(
+      final Extension value, final JsonGenerator gen, final SerializerProvider provider) throws IOException
+  {
+    if (value.getExtensionType().equals(ExtensionType.VULNERABILITIES)) {
+      if (gen instanceof ToXmlGenerator) {
+        try {
+          serializeVulnerabilities((ToXmlGenerator) gen, value);
+        }
+        catch (XMLStreamException e) {
+          e.printStackTrace();
+          throw new IOException(e);
+        }
+      }
+    }
+  }
+
+  private void serializeVulnerabilities(final ToXmlGenerator gen, final Extension vulns)
+      throws XMLStreamException
+  {
+    final XMLStreamWriter staxWriter = gen.getStaxWriter();
+    staxWriter.writeStartElement(
+        vulns.getPrefix(),
+        "vulnerabilities",
+        vulns.getNamespaceURI());
+    for (ExtensibleType ext : vulns.getExtensions()) {
+      final Vulnerability1_0 vuln = (Vulnerability1_0) ext;
+      staxWriter.writeStartElement(vulns.getPrefix(), "vulnerability", vulns.getNamespaceURI());
```

Comment:
Idea: since we have those values here and in the deserializer and in the new class for vulnerability, we could have them as constants somewhere, so if any of those change we don't have to change three classes but only one

Reply:
Thankfully I already did this, I just didn't use them:

```
  public static final String NAMESPACE_URI = "http://cyclonedx.org/schema/ext/vulnerability/1.0";
  public static final String PREFIX = "v";
  public static final String NAME = "vulnerability";
```

lol

Reply:
> Thankfully I already did this, I just didn't use them:
> 
> ```
>   public static final String NAMESPACE_URI = "http://cyclonedx.org/schema/ext/vulnerability/1.0";
>   public static final String PREFIX = "v";
>   public static final String NAME = "vulnerability";
> ```
> 
> lol

Good, I'd like to see it for the other attributes of vulnerabilities as well, if you check they are repeated everywhere

Reply:
8052b82

Reply:
Remember this, for the rest of your days: aa9ac0d